### PR TITLE
Implement setTensorValue in InputModelPDPD

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Implement setTensorValue in InputModelPDPD*

### Tickets:
 - *53097*

### CODE REVIEW

**Comment by nosovmik** [comment]
Can you please add unit test for this new method? It should not be too complicated. And it will be helpful for developers to understand what kind of format "const void*" should be

File: ngraph/frontend/paddlepaddle/include/paddlepaddle_frontend/model.hpp [hunk 1]
```diff
@@ -18,11 +18,14 @@
 
 #include <frontend_manager/frontend_manager.hpp>
 
-#include "place.hpp"
+//#include "place.hpp"
```
Comment (mvafin) [inline ngraph/frontend/paddlepaddle/include/paddlepaddle_frontend/model.hpp:21]: need to remove

**Comment by mvafin** [comment]
@nosovmik Implemented unit-test for setTensorValue

File: ngraph/test/frontend/shared/src/cut_specific_model.cpp [hunk 2]
```diff
@@ -216,3 +216,21 @@ TEST_P(FrontEndCutModelTest, testExtractSubgraph) {
...
+    ASSERT_NO_THROW(place = m_inputModel->getPlaceByTensorName(m_param.m_tensorValueName));
+    ASSERT_NO_THROW(m_inputModel->setTensorValue(place, &m_param.m_tensorValue[0]));
+
+    std::shared_ptr<ngraph::Function> function;
+    ASSERT_NO_THROW(function = m_frontEnd->convert(m_inputModel));
+    auto ops = function->get_ordered_ops();
+
+    // Ensure that it doesn't contain old input
+    auto name = m_param.m_op_before_name;
+    EXPECT_TRUE(std::find_if(ops.begin(), ops.end(),
```
Comment (nosovmik) [inline ngraph/test/frontend/shared/src/cut_specific_model.cpp:232]: It's not clear how 'setTensorValue' correlates with expectation of not having old input...
Can we check instead that tensor value of converted ngraph::Function matches with m_param.m_tensorValue[0]?

Comment (mvafin) [reply ngraph/test/frontend/shared/src/cut_specific_model.cpp]: Done

File: ngraph/test/frontend/shared/src/cut_specific_model.cpp [hunk 3]
```diff
@@ -6,6 +6,7 @@
 #include <algorithm>
 #include "../include/cut_specific_model.hpp"
 #include "../include/utils.hpp"
+#include "../../../../core/include/ngraph/opsets/opset7.hpp"
```
Comment (nosovmik) [inline ngraph/test/frontend/shared/src/cut_specific_model.cpp:9]: could be just <ngraph/opsets/opset7.hpp>

File: ngraph/test/frontend/shared/include/cut_specific_model.hpp [hunk 4]
```diff
@@ -16,6 +16,9 @@ struct CutModelParam {
     std::vector<std::string> m_newInputs;
     std::vector<std::string> m_oldOutputs;
     std::vector<std::string> m_newOutputs;
+    std::string              m_tensorValueName;
```
Comment (nosovmik) [inline ngraph/test/frontend/shared/include/cut_specific_model.hpp:19]: I believe we should consider to create separate test files for this, like 'tensor_value.cpp'. Per my understanding, this test is not related to 'cutting model' functionality.

File: ngraph/test/frontend/shared/src/cut_specific_model.cpp [hunk 5]
```diff
@@ -216,3 +217,23 @@ TEST_P(FrontEndCutModelTest, testExtractSubgraph) {
...
+    auto ops = function->get_ordered_ops();
+
+    auto const_name = m_param.m_tensorValueName;
+    auto const_node_it = std::find_if(ops.begin(), ops.end(),
+                                   [&](const std::shared_ptr<ngraph::Node>& node) {
+                                       return node->get_friendly_name().find(const_name) != std::string::npos;
+                                   });
+    ASSERT_TRUE(const_node_it != ops.end()) << "Name shall exist:" << const_name;
+    auto data = std::dynamic_pointer_cast<opset7::Constant>(*const_node_it)->get_vector<float>();
+    EXPECT_TRUE(std::equal(data.begin(), data.end(), m_param.m_tensorValue.begin())) << "Data must be equal";
```
Comment (nosovmik) [inline ngraph/test/frontend/shared/src/cut_specific_model.cpp:238]: Also would be good to add 'reverse' check, like
EXPECT_TRUE(std::equal(m_param.m_tensorValue.begin(), m_param.m_tensorValue.end(), data.begin())) << "Data must be equal";
Otherwise - when 'data' is empty test will be passed.
Alternatively, you can add check of EXPECT_EQ(data.size(), m_param.m_tensorValue.size())

File: ngraph/test/frontend/shared/include/cut_specific_model.hpp (hunk 4 above)
Comment (mvafin) [reply ngraph/test/frontend/shared/include/cut_specific_model.hpp:19]: It is related to cutting the model. When you specify a tensor value in the middle of the model you cut everything that was before this tensor.

**Review by nosovmik — APPROVED** [review]
